src/test/java/bio/terra/endpoints/CLICommandTests.java

```diff
@@ -0,0 +1,284 @@
+     */
+    public Map<String, Object> datasetCreateTest(String token, String inputJSONFilename, String expectedOutputFilename)
+            throws IOException {
+        System.out.println("***********************************************");
+        System.out.println("jc dataset create --input-json inputDatasetCreate.txt");
+
+        // build path to input file
+        String inputJSON = CLITestingConfig.dirName + inputJSONFilename;
+
+        // call CLI command in a separate process
+        List<String> cmdArgs = new ArrayList<String>(new ArrayList<>(Arrays.asList(
+                new String[]{"dataset", "create", "--input-json", inputJSON})));
+        List<String> cliCmdResponse = CLITestingUtils.callCLICommand(cmdArgs);
+
+        // read in the expect CLI command output from a file
+        List<String> cliCmdExpectedResponse = CLITestingUtils.readCLIExpectedOutput(expectedOutputFilename);
+
+        // log the response to stdout
+        System.out.println("cliCmdResponse: " + cliCmdResponse + "\n");
+
+        // fetch the dataset summary and details over HTTP
+        Map<String, Object> inputJSONMap = CLITestingUtils.readCLIInput(inputJSONFilename);
+        Map<String, Object> datasetSummary = enumerateDatasetHttpGET(token, inputJSONMap.get("name").toString());
+        Map<String, Object> datasetDetails = retrieveDatasetHttpGET(token, datasetSummary.get("id").toString());
+
+        // need to replace %xyz% variables in the CLI expected response with values from the Java HTTP response
+        String id = datasetSummary.get("id").toString();
+        String createdDate = datasetSummary.get("createdDate").toString();
+
+        // check that the responses match
+        for (int ctr = 0; ctr < cliCmdResponse.size(); ctr++) {
+            String expectedLine = cliCmdExpectedResponse.get(ctr);
+            expectedLine = expectedLine.replace("%id%", id);
+            expectedLine = expectedLine.replace("%createdDate%", createdDate);
+
+            Assert.assertEquals(expectedLine, cliCmdResponse.get(ctr));
+        }
+
+        return datasetDetails;
+    }
+
+    /**
+     * HTTP GET request : GET : enumerateDatasets
+     * @param token access token to send with request
+     * @return dataset summary as a property map
+     * @throws IOException
+     */
+    public Map<String, Object> enumerateDatasetHttpGET(String token, String datasetName) throws IOException {
+        System.out.println("***********************************************");
+        System.out.println("HTTP request api/repository/v1/datasets");
+
+        // endpoint information
+        String endpointName = "api/repository/v1/datasets";
+        String endpointType = "GET";
+
+        // build request parameter map
+        Map<String, String> params = new HashMap<>();
+        params.put("filter", datasetName);
+
+        // make request using Java HTTP library
+        Map<String, Object> javaHttpResponse = CLITestingUtils.sendJavaHttpRequest(
```

Comment:
Why use raw HTTP? You have the codegen API available.

Reply:
Yes I'm curious about this as well. What do we gain from being at this lower level?

Reply:
I wanted the tests to be as independent of the code as possible. I think it makes them easier to debug if they don't depend on classes in src, also easier to reproduce because you don't need the rest of the repository. This general guideline was really helpful at my last job working on a JDBC driver, but maybe that's not as applicable for our application, since it's higher up the stack. What do you think?